Examples/ResourceLeak/README.md

```diff
@@ -0,0 +1,50 @@
+## Procedure-local Bugs of Resource Leaks
+
+**CSharpCodeAnalyzer** is designed to detect this kind of resource leaks by applying the following principle: **When a new object is allocated during the execution of a procedure, it is the procedure’s responsibility to either deallocate the object or make it available to its callers; there is no such obligation for objects received from the caller.** 
+
+Based on the nature of procedure-local bugs of resource leaks, **CSharpCodeAnalyzer** is capable of detecting resource leaks in the following scenarios that are unrelated to multi-threading:
+
+## Supported scenarios
+
+### 1. Standard idioms: 
+
+Some objects in C#, the resources, are supposed to be disposed when you stop using them, and failure to dispose is a resource leak. Resources include input streams, output streams, readers, writers, sockets, http connections, cursors, json parsers, etc. The following code snippet shows a *ReaderStream* object is created and disposed. 
+
+```c#
+public void ResourceLeakIntraproceduralOK(){
+    string data;
+    StreamReader sr = new StreamReader("whatever.txt");            
+    data = sr.ReadToEnd();
+    sr.Close();
+}
+```
+							
+### 2. Nested allocations: 
+When a resource allocation is included as an argument to a constructor, if the constructor fails it can leave an an unreachable resource that no one can dispose.
+	
+The following example 
+```c#
+var gzipStream = new GZipStream(new FileStream(out, FileMode.Create), CompressionMode.Compress);
+```
+is bad in case the outer constructor, *GZipStream*, throws an exception. In that case, no one will have a hold of the *FileStream* and so no one will be able to dispose it.
```

Comment:
try `GZipStream` to see which one looks better

Reply:
use ``

Reply:
Resolved in the update